Compile fails on Rust 1.38 and 1.39

`cfg(doctest)` implemented in Rust 1.38 as an unstable feature, and stabilized in Rust 1.40. It compiles fine on Rust 1.37 and older, but fails on Rust 1.38 and 1.39, where this feature existed as an unstable feature.

https://github.com/BurntSushi/termcolor/blob/b052b61ba052c7e5672bb91516982d27cbf9129d/src/lib.rs#L122-L125

Error:
```console
$ cargo +1.39.0 check
    Checking termcolor v1.1.1 (/Users/taiki/projects/termcolor)
error[E0658]: `cfg(doctest)` is experimental and subject to change
   --> src/lib.rs:122:7
    |
122 | #[cfg(doctest)]
    |       ^^^^^^^
    |
    = note: for more information, see https://github.com/rust-lang/rust/issues/62210
```

cc https://github.com/BurntSushi/termcolor/pull/31
Related: https://github.com/rust-lang/regex/issues/685

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -22,4 +22,6 @@ bench = false
 winapi-util = "0.1.3"
 
 [dev-dependencies]
-doc-comment = "0.3"
+# TODO: Re-enable this once the MSRV is 1.43 or greater.
+# See: https://github.com/BurntSushi/termcolor/issues/35
+# doc-comment = "0.3"

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -119,10 +119,10 @@ Currently, `termcolor` does not provide anything to do this for you.
 
 #![deny(missing_docs)]
 
-#[cfg(doctest)]
-use doc_comment::doctest;
-#[cfg(doctest)]
-doctest!("../README.md");
+// #[cfg(doctest)]
+// use doc_comment::doctest;
+// #[cfg(doctest)]
+// doctest!("../README.md");
 
 use std::env;
 use std::error;

